hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -0,0 +1,79 @@
+package org.apache.iceberg.hive;
+
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.thrift.TException;
+
+import static org.apache.iceberg.BaseMetastoreTableOperations.METADATA_LOCATION_PROP;
+
+public class HiveCatalog extends BaseMetastoreCatalog {
+
+  private final HiveClientPool clients;
+  private final Configuration conf;
+
+  public HiveCatalog(Configuration conf) {
+    super(conf);
+    this.conf = conf;
+    this.clients = new HiveClientPool(2, conf);
+  }
+
+  @Override
+  public void dropTable(TableIdentifier tableIdentifier) {
+    validateTableIdentifier(tableIdentifier);
+    HiveMetaStoreClient hiveMetaStoreClient = this.clients.newClient();
+    try {
+      hiveMetaStoreClient.dropTable(tableIdentifier.namespace().levels()[0], tableIdentifier.name());
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + tableIdentifier.toString(), e);
+    }
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    validateTableIdentifier(from);
+    validateTableIdentifier(to);
+
+    HiveMetaStoreClient hiveMetaStoreClient = this.clients.newClient();
+    String location = ((BaseTable) getTable(from)).operations().current().file().location();
```

Comment:
I don't think renaming the table should change the location. Data can be stored in the old location and referenced by a new name.